test/pylib/scylla_cluster.py

```diff
@@ -100,7 +104,7 @@ def make_scylla_conf(workdir: pathlib.Path, host_addr: str, seed_addrs: List[str
 # it easier to restart. Sic: if you make a typo on the command line,
 # Scylla refuses to boot.
 SCYLLA_CMDLINE_OPTIONS = [
-    '--smp', '2',
+    '--smp', str(NUM_SHARDS),
```

Comment:
Can't you reset it from the configuration?

Reply:
suite.yaml allows you to reset it.

Reply:
Hmm, if `--smp` it's overridden by suite options, I think it should not appear here at all.

Reply:
But where?
```
$ gg "smp" test/pylib/
test/pylib/scylla_cluster.py:    '--smp', str(NUM_SHARDS),
$ gg "smp" test.py
$ 
```

Reply:
Ah, `extra_scylla_cmdline_options`.

Reply:
Actually, it doesn't work
```yaml
extra_scylla_cmdline_options: ['--smp', '3']
```

results in:
```
error: option '--smp' cannot be specified more than once
```

that's because the options are concatenated:
```python3
        scylla_args = SCYLLA_CMDLINE_OPTIONS + self.cmdline_options
```

so I guess we're good :D

Reply:
we're not good. this will definitely be used. if it broke, it shouldn't break even more

Reply:
It's not breaking even more, passing smp from suite.yaml or otherwise is now impossible, IMO fixing that is out of this PR's scope.

There should be a separate PR for fixing that - when we actually start testing clusters with different `smp`s. And I think it should be more flexible than a suite option, we should be able to restart nodes with different `smp`s and so on, it's a bigger refactor. In any case, I see no reason why it should block this PR.

Reply:
I've [cooked a general syntax](https://github.com/scylladb/scylladb/pull/12374/files#diff-a618a6143ed1c1bb9ef60beab017e6b34bb3919737232341ff36893b25ed6958R125) for setting command line options, maybe we should use it here.